apps/OboeTester/app/src/main/java/com/mobileer/audio_device/AudioDeviceInfoConverter.java

```diff
@@ -74,30 +76,37 @@ public static String toString(AudioDeviceInfo adi){
             sb.append(adi.getAddress());
         }
 
-        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.R) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
             sb.append("\nEncapsulation Metadata Types: ");
             int[] encapsulationMetadataTypes = adi.getEncapsulationMetadataTypes();
             sb.append(intArrayToString(encapsulationMetadataTypes));
         }
 
-        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.R) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
             sb.append("\nEncapsulation Modes: ");
             int[] encapsulationModes = adi.getEncapsulationModes();
             sb.append(intArrayToString(encapsulationModes));
         }
 
-        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.S) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
             sb.append("\nAudio Descriptors: ");
             List<AudioDescriptor> audioDescriptors = adi.getAudioDescriptors();
             sb.append(audioDescriptors);
         }
 
-        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.S) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
             sb.append("\nAudio Profiles: ");
             List<AudioProfile> audioProfiles = adi.getAudioProfiles();
             sb.append(audioProfiles);
         }
 
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+            sb.append("\nSupported Mixer Attributes: ");
```

Comment:
It may only be needed if it is a USB output device. For all other kinds of device, it will just be empty.

Reply:
It's fine if we output an empty array on the other devices. It's better for debugging if we write [] over not writing anything